martin/src/config/file/main.rs

```
@@ -365,6 +393,38 @@ impl Config {
     }
 }
 
+#[derive(PartialEq, Eq, Debug, Clone, Copy, Default, Serialize, Deserialize, ValueEnum)]
+#[serde(rename_all = "lowercase")]
+pub enum OnInvalid {
+    /// Print warning message, and abort if the error is critical
+    #[default]
+    Warn,
+    /// Abort startup if any warnings
+    Abort,
+    /// Ignore all warnings
+    Ignore,
+}
+
+impl OnInvalid {
+    /// Handle warnings based on the configured behavior
+    pub fn handle_warnings(policy: Self, warnings: &[TileSourceWarning]) -> MartinResult<()> {
+        if warnings.is_empty() {
+            return Ok(());
+        }
+        match policy {
+            OnInvalid::Abort => Err(MartinError::TileResolutionWarningsIssued),
+            OnInvalid::Warn => {
+                warn!("Tile source resolution warnings:");
+                for warning in warnings {
+                    warn!("  - {warning}");
+                }
+                Ok(())
+            }
+            OnInvalid::Ignore => Ok(()),
+        }
+    }
```

thee requests on the current code:

- (optional) if there is only one warning, lets add a special case with better log formatting
- `OnInvalid::Abort` should log at error level (currently this swallows the log)
- `OnInvalid::Ignore` should likely log at debug or trace level (-> better debuggability)